teemaderegister-react/src/actions/AdminActions.js

```diff
@@ -0,0 +1,23 @@
+import Api from '../utils/Api'
+import * as types from '../constants/ActionTypes'
+import { setToken } from '../utils/jwt'
+
+import { ADMIN_ADD_NEW_USER } from '../constants/ApiConstants'
+
+export const initAddNewUser = () => dispatch => dispatch({ type: types.CREATE_USER_INIT })
+
+export const adminAddNewUser = userData => {
```

Comment:
there are two ways to approach this, one is to use reducers (check CurriculumReducers as example). Another is to user responses directly. 

You are using responses directly, so actually there is no need to use dispatch any actions....

If You want to use reducers then instead of this later
```js
const response = await props.adminAddNewUser(values)
    const {message, success} = response

    if (success === 1) {
      notification.success({message: message})
      form.resetFields()
    } else {
      notification.error({message: message})
    }
```

You should be using this + useEffect to trigger something on props changes as response comes from props
```
props.adminAddNewUser(values)
```

Reply:
So I would leave up to You. Use reducers or leave it as is but cleanup the actions as You are not actually using them.